Update changelog and versions for v2.12-beta.1 (#12466)
mapbox/mapbox-gl-js: CHANGELOG.md

## 2.12.0-beta.1

## Features ✨ and improvements 🏁

> * Improve performance of patterns and line dashes, also improving their appearance when zooming ([#12326](https://github.com/mapbox/mapbox-gl-js/pull/12326))

Reviewer: ```suggestion
* Improve performance of patterns and line dashes and improve their appearance when zooming. ([#12326](https://github.com/mapbox/mapbox-gl-js/pull/12326))
```

* Improve performance of text and icon placement. ([#12351](https://github.com/mapbox/mapbox-gl-js/pull/12351))
* Improve performance of loading terrain data ([#12397](https://github.com/mapbox/mapbox-gl-js/pull/12397))

> * Improve behavior of `getBounds` at the poles on a globe projection. ([#12315](https://github.com/mapbox/mapbox-gl-js/pull/12315))

Reviewer: Should we consider this a bugfix?

Author: Yes, let's move that one on bugfix.

> * Improve poles support in `getBounds` when projection is set to globe ([#12315](https://github.com/mapbox/mapbox-gl-js/pull/12315))

Reviewer: ```suggestion
* Improve behavior of `getBounds` at the poles on a globe projection. ([#12315](https://github.com/mapbox/mapbox-gl-js/pull/12315))
```

Reviewer: Also, should we consider this a bugfix?

> * Allow zooming towards terrain at a safe distance without pitching the camera ([#12354](https://github.com/mapbox/mapbox-gl-js/pull/12354))

Reviewer: Should this be a bugfix? Maybe something like:
```
Fix pitch changing to 0 on zooming in to terrain
```

Author: I was also considering moving it to a bug fix but it seems more fitting under the improvement section as the "right" behavior can be quite subjective.

Reviewer: Good point! My thinking was that it fixed [this bug](https://github.com/mapbox/mapbox-gl-js/issues/10197) but it looks like the scope of that ticket is smaller. Thoughts on rephrasing to:

```suggestion
* Update collisions with terrain to longer adjust pitch and reduce clipping. Mouse panning into terrain now lifts the camera above terrain. ([#12354](https://github.com/mapbox/mapbox-gl-js/pull/12354))
```

cc @avpeery

Reviewer 2: I personally considered it a bug but seems like both phrasings and categories fit, so whatever works!

> * Allow for pitch override in cameraForBounds, `fitBounds` and `fitScreenCoordinates` camera APIs ([#12367](https://github.com/mapbox/mapbox-gl-js/pull/12367))

Reviewer: ```suggestion
* Allow for pitch override in `cameraForBounds`, `fitBounds` and `fitScreenCoordinates` camera APIs. ([#12367](https://github.com/mapbox/mapbox-gl-js/pull/12367))
```
